File: BLH.ApproveIQ.Application/DTOs/InvoiceListDto.cs
namespace BLH.ApproveIQ.Application.DTOs;

public class InvoiceListDto
{
    public Guid Id { get; set; }
    public string? InvoiceNumber { get; set; }
    public DateTime? InvoiceDate { get; set; }
    public string? VendorName { get; set; }
    public string? VendorId { get; set; }
    public string? ProjectName { get; set; }         // From Project.Name
    public string? BuJobNumber { get; set; }
    public decimal? GrossAmount { get; set; }
    public string? InvoiceStatus { get; set; }
    
    // Assigned user info
    public string? AssignedUserFirstName { get; set; }  // From CurrentlyAssignedToUser.FirstName
    public string? AssignedUserLastName { get; set; }   // From CurrentlyAssignedToUser.LastName
    public string? AssignedUserFullName { get; set; }   // Computed field for frontend
}

File: BLH.ApproveIQ.Portal/ClientApp/src/app/data/data/b-l-h/dtos/invoice-list-dto.ts
export class InvoiceListDto {
    id: string;
    invoiceNumber: string;
    invoiceDate = new Date();
    vendorName: string;
    vendorId: string;
    projectName: string;
    buJobNumber: string;
    grossAmount?: number;
    invoiceStatus: string;
    assignedUserFirstName: string;
    assignedUserLastName: string;
    assignedUserFullName: string;
}
